nginx/modsecurity/ngx_http_modsecurity.c

```diff
@@ -953,14 +962,21 @@ ngx_http_modsecurity_init(ngx_conf_t *cf)
     }
     *h = ngx_http_modsecurity_handler;
 
-    ngx_http_next_header_filter = ngx_http_top_header_filter;
-    ngx_http_top_header_filter = ngx_http_modsecurity_header_filter;
-
-    ngx_http_next_body_filter = ngx_http_top_body_filter;
-    ngx_http_top_body_filter = ngx_http_modsecurity_body_filter;
-
     ngx_memzero(&ngx_http_modsecurity_upstream, sizeof(ngx_http_upstream_t));
     ngx_http_modsecurity_upstream.cacheable = 1;
+
+    ngx_thread_pool_t* thread_pool = ngx_thread_pool_add(cf, &thread_pool_name);
+    if (thread_pool == NULL) {
```

Comment:
Can we add log messages to explain the different errors?

Reply:
I looked at the implementation of ngx_thread_pool_add. It just returns NULL if the internal state was somehow broken. It doesn't return any indication of what was wrong either, so we don't really have any way to know. I think it'll be more of a programming error than an error that could happen at runtime, so I'm not too worried.

Reply:
Can we log messages for the thread pool part? "Thread pool could not be created" also works...